src/views/ApartmentApp/components/StudentApplication/index.js

```diff
@@ -640,29 +612,14 @@ const StudentApplication = ({ userProfile, authentication }) => {
     }
   };
 
-  /**
-   * Callback for apartment application submit button
-   */
-  const handleSubmitButtonClick = () => {
-    let saveResult = saveApartmentApplication(applicationDetails);
-    if (saveResult) {
-      setSubmitDialogOpen(true);
-    }
-  };
-
-  const handleSubmitAppAccepted = () => {
-    // The method is separated from callback because the housing API service must be handled inside an async method
-    submitApplication();
-    handleCloseOkay();
-  };
-
   /**
    * Submit the current application as completed
    *
    * @async
    * @function submitApplication
    */
   const submitApplication = async () => {
+    handleCloseDialog();
```

Comment:
It's a little strange to have the dialog behavior coupled this tightly with the data handling. Would it be reasonable to have the `handleCloseDialog()` call in an anonymous callback that also calls `submitApplication()`?

Reply:
It's used to be in a callback, which I shorten and made in-line, 
So there are several ways of moving it back

Reply:
I meant to convey that I do agree, it feels inappropriate to lump the dialog stuff into the operation of "submitting" or "saving" or whatever else